packages/hardhat-core/src/internal/core/jsonrpc/types/base-types.ts

```diff
@@ -40,6 +41,14 @@ export const rpcStorageSlot = new t.Type<bigint>(
   t.identity
 );
 
+export const rpcStorageSlotHexStr = new t.Type<string>(
+  "Storage slot hex string",
+  isString,
```

Comment:
We try to avoid using lodash because we'll probably remove it as a dependency in the future. In this particular case, `x => typeof x === "string"` is enough for our purposes. Lodash's implementations also accept String objects, but that's not relevant for us.